tests/conftest.py

```diff
@@ -74,12 +74,58 @@
 asyncio.set_event_loop_policy(AnyThreadEventLoopPolicy())
 logger = logging.getLogger(__name__)
 
+TABLES_TO_KEEP = [x.table_name() for x in data._classes]
+
+# Database
+@pytest.fixture
+def postgres_proc(unused_tcp_port_factory):
+    proc = PostgresProc(unused_tcp_port_factory())
+    yield proc
+    proc.stop()
+
 
 @pytest.fixture(scope="session", autouse=True)
 def postgres_db(postgresql_proc):
     yield postgresql_proc
 
 
+@pytest.fixture(scope="function")
+async def create_db(postgres_db, database_name_internal):
+    """
+    see :py:database_name_internal:
+    """
+    connection = await asyncpg.connect(host=postgres_db.host, port=postgres_db.port, user=postgres_db.user)
+    try:
+        await connection.execute("CREATE DATABASE " + database_name_internal)
+    except DuplicateDatabaseError:
+        pass
+    finally:
+        await connection.close()
+    return database_name_internal
+
+
+@pytest.fixture(scope="session")
+def database_name_internal():
+    """
+    The database_name fixture is expected to yield the database name to an existing database, and should be session scoped.
+
+    However, async fixtures all depend on the event loop, which is function scoped.
+    To create the database, we need asyncpg, which needs an eventloop.
+
+    To resolve this, there is a session scoped fixture called database_name_internal that provides a fixed name,
+    but can not ensure the database has been created.
+
+    and a database_name fixtures that does ensure the database is created, but this forces it to be function scoped
```

Comment:
The reasoning is clear now, except for this sentence. Can you fix this sentence?